Source/Libraries/FaultData/DataOperations/DFRLineFiles/LineFileLoader.cs

```diff
@@ -0,0 +1,681 @@
+            int lineSegmentID = m_connection.ExecuteScalar<int>("SELECT ChildID FROM AssetRelationship WHERE AssetRelationshipTypeID = {0} AND ParentID = {1}", assetRelationshipType, line.ID);
+            TableOperations<LineSegment> lineSegmentTable = new TableOperations<LineSegment>(m_connection);
+            LineSegment lineSegment = lineSegmentTable.QueryRecordWhere("ID = {0}", lineSegmentID);
+            MeterAsset meterAsset = null;
+
+            if (lineSegment is null)
+            {
+                string assetKey = $"{line.AssetKey}-Segment1";
+
+                lineSegment = new LineSegment()
+                {
+                    AssetKey = assetKey,
+                    AssetName = $"{lineSection.LineName} - Segment 1",
+                    Length = lineSection.Length,
+                    R0 = lineSection.R0,
+                    X0 = lineSection.X0,
+                    R1 = lineSection.R1,
+                    X1 = lineSection.X1,
+                };
+
+                try
+                {
+                    lineSegmentTable.AddNewRecord(lineSegment);
+                    lineSegment.ID = m_connection.ExecuteScalar<int>("SELECT ID FROM Asset WHERE AssetKey = {0}", assetKey);
+
+                    AssetConnection assetConnection = new AssetConnection()
+                    {
+                        AssetRelationshipTypeID = assetRelationshipType,
+                        ParentID = line.ID,
+                        ChildID = lineSegment.ID
+                    };
+
+                    TableOperations<AssetConnection> assetConnectionTable = new TableOperations<AssetConnection>(m_connection);
+                    assetConnectionTable.AddNewRecord(assetConnection);
+
+                    meterAsset = new MeterAsset()
+                    {
+                        MeterID = meter.ID,
+                        AssetID = line.ID
+                    };
+
+                    TableOperations<MeterAsset> meterAssetTable = new TableOperations<MeterAsset>(m_connection);
+                    meterAssetTable.AddNewRecord(meterAsset);
+                    meterAsset.ID = m_connection.ExecuteScalar<int>("SELECT ID FROM MeterAsset WHERE MeterID = {0} AND AssetID = {1}", meter.ID, line.ID);
+                }
+                catch (SqlException ex)
+                {
+                    // Ignore errors regarding unique key constraints
+                    // which can occur as a result of a race condition
+                    bool isUniqueViolation = (ex.Number == 2601) || (ex.Number == 2627);
+
+                    if (!isUniqueViolation)
+                        throw;
+                }
+            }
+            else if (lineSegment.Length != lineSection.Length || lineSegment.R0 != lineSection.R0 || lineSegment.X0 != lineSection.X0 || lineSegment.R1 != lineSection.R1 || lineSegment.X1 != lineSection.X1)
```

Comment:
My current thinking is that users shouldn't be using DFR line files to manage lines with information about multiple segments. Therefore, I can't really think of a valid use-case to model how best to handle this situation. Essentially, if the situation would ever come up, it represents a misunderstanding of how the system works, and some adjustments will need to be made by the users to reach whatever outcome it was that they were expecting. In other words, I don't really see how one way of handling this case provides any real advantage over any other way of handling it.

For the general case, however, I would much prefer updating existing records instead of deleting and re-adding because it reduces churn on the autoincrementing primary keys. Unlike FAWG which updates once a month, this data operation executes automatically each time we process from a meter so I feel it would be prudent to keep the configuration data that it manages as stable as possible.